pkg/apply/task/apply_task.go

```diff
@@ -116,6 +127,22 @@ func (a *ApplyTask) Start(taskContext *taskrunner.TaskContext) {
 					object.UnstructuredToObjMeta(obj), event.Unchanged, applyerror.NewUnknownTypeError(err))
 				continue
 			}
+
+			clusterObj, err := getClusterObj(dynamic, info)
+			if err != nil {
+				if !apierrors.IsNotFound(err) {
+					fmt.Println("I'm here in line 134")
+					if !inventory.CanApply(a.InvInfo, clusterObj, a.InventoryPolicy) {
+						taskContext.EventChannel() <- createApplyEvent(
+							object.UnstructuredToObjMeta(obj),
+							event.Unchanged,
+							applyerror.NewCannotApplyError(fmt.Errorf("can't apply the resource since it belongs to another inventory object")))
```

Comment:
Is this error message always correct? If the policy is `InventoryPolicyMustMatch`, apply can be blocked simply because the resource already exist, even if it doesn't belong to another inventory.

Reply:
That's true. We can let `CanApply` return a message for why the resource can't be applied. Then this error will capture that message.